starky/src/deserializer.rs

```diff
@@ -0,0 +1,400 @@
+                    //     pub last: Vec<F>,
+                    // }
+                    // stark_proof.evals = Some(map.next_value()?);
+
+
+                }
+                "s0_vals3" => {
+                    let s0_vals3: Vec<Vec<F3G>> = map.next_value()?;
+
+                    // stark_proof.evals = Some(map.next_value()?);
+                }
+                "s0_vals4" => {
+                    let s0_vals4: Vec<Vec<F3G>> = map.next_value()?;
+
+                    // stark_proof.evals = Some(map.next_value()?);
+                }
+                "s0_valsC" => {
+                    let s0_valsC: Vec<Vec<F3G>> = map.next_value()?;
+
+                    // stark_proof.evals = Some(map.next_value()?);
+                }
+                "s0_siblings1" => {
+                    let s0_siblings1: Vec<Vec<Vec<Input<M::MTNode>>>> = map.next_value()?;
+
+                    // stark_proof.evals = Some(map.next_value()?);
+                }
+                "s0_siblings2" => {
+                    let s0_siblings2: Vec<Vec<Vec<Input<M::MTNode>>>> = map.next_value()?;
+
+                    // stark_proof.evals = Some(map.next_value()?);
+                }
+                "s0_siblings3" => {
+                    let s0_siblings3: Vec<Vec<Vec<Input<M::MTNode>>>> = map.next_value()?;
+
+                    // stark_proof.evals = Some(map.next_value()?);
+                }
+                "s0_siblings4" => {
+                    let s0_siblings4: Vec<Vec<Vec<Input<M::MTNode>>>> = map.next_value()?;
+
+                    // stark_proof.evals = Some(map.next_value()?);
+                }
+                "s0_siblingsC" => {
+                    let s0_siblingsC: Vec<Vec<Vec<Input<M::MTNode>>>> = map.next_value()?;
+
+                    // stark_proof.evals = Some(map.next_value()?);
+                }
+            }
+        }
+
+
+        {
+            let hashtype = &self.stark_struct.verificationHashType;
+            match &self.rootC {
+                Some(value) => {
+                    map.serialize_entry("rootC", &Input::<M::MTNode>::new(*value, hashtype.clone()))?;
+                }
+                None => {}
+            }
+
+            map.serialize_entry(
+                "root1",
+                &Input::<M::MTNode>::new(self.root1, hashtype.clone()),
+            )?;
+            map.serialize_entry(
+                "root2",
+                &Input::<M::MTNode>::new(self.root2, hashtype.clone()),
+            )?;
+            map.serialize_entry(
+                "root3",
+                &Input::<M::MTNode>::new(self.root3, hashtype.clone()),
+            )?;
+            map.serialize_entry(
+                "root4",
+                &Input::<M::MTNode>::new(self.root4, hashtype.clone()),
+            )?;
+            map.serialize_entry("evals", &self.evals)?;
+        }
+
+
+        // TODO: Below are the serilized one.
+
+        for i in 1..(self.fri_proof.queries.len()) {
```

Comment:
Here need to know pol_queries_len, but can't.

Reply:
We can get the lens from starkStruct.json, you can pass the file into deserializer